image/unparsed.go

```diff
@@ -15,6 +15,9 @@ type UnparsedImage = image.UnparsedImage
 // UnparsedInstance returns a types.UnparsedImage implementation for (source, instanceDigest).
 // If instanceDigest is not nil, it contains a digest of the specific manifest instance to retrieve (when the primary manifest is a manifest list).
 //
+// This implementation of [types.UnparsedImage] ensures that [types.UnparsedImage.Manifest] validates the image
+// against instanceDigest if set, or, if not, a digest implied by src.Reference, if any.
```

Comment:
The “implied by” is intentionally _a bit_ vague, there is #1049 and I’m not yet sure how to handle that. For now, we must provide this feature for at least registry references, and in that case the meaning is clear enough.